### ISSUE TITLE
[BUG] Just Flight PA28 issues

### ISSUE THREAD

**Issue by CptAdhoc** [Sequal32/yourcontrols#96]
Hello,

I have made my first .yaml file yesterday with the PA28 few hours before the beta release :)
During my own tests, I found some issues that I tried to reproduce with your version.
Here is a list of what i found so far, and how I solve that on my side.
I attach my own .yaml file, feel free to use it. Howevere, since it's my thirst attempt, it isn't perfect, it wasn't fully test, fuel synchronisation is missing, no equipement visibility variable (L:[...]_VIZ) and COM1/2 NAV1/2 on the KR170/175 don't sync.

- **Door (oil, bagage, window, passenger door won't open. There is a flikering movement but return to there initial position**. No solution for that, I guess moving the door require more than just adjusting door variables.

- ** Flaps not working, the flap lever move but return to his initial position.** I used "L:LOWER_flap_handle". I removed the flap section of "general.yaml" (quick and dirty way), it works with that but the lever do some back and forth before going to the good position.

- **Fuel selector not working**. I used "L:LEFT_MISC_fuel_sel"

- **KMA20 (audio switchs) not working.** See my .yaml for all "L.KMA20_[...] in #KMA section.

- **KX170 / KX175 COM-NAV on/off not working**. I used "L:KX170_COMM1_offon" and "L:KX175_COMM2_offon". Watch out, com1 is KX170, com2 KX175.

- **ADF frequency jump digits**.  Maybe it's due to the utilisation of both "KR85_[...]_knob" and "L:KR85_digXXX_counter". Using only counter variable seems to work.

- **Transponder STBY/ON/IDENT and IDENT button not working**. I used "L:KT76_mode_knob" ; "L:KT76_ident_button".

- **DME no working.** I used "L:KN62_power" ; "L:KN62_mode_knob"

- **Direction indicator drift knob not working**. I used "L:DIRECTION_INDICATOR_card_knob"

- **ELT not working**. I used "L:LEFT_MISC_elt".

- **Autopilote mode knob and nav selector not working**. I used "L:AUTOPILOT_mode" ; "L:AUTOPILOT_nav"

There is probably more, the PA28 seems to include a lot of local variable (at least for my newby skills :D and as far as I understand how YC works). Again, feel free to use my .yaml file for reference.
[JustFlight_PA28_Arrow_III(custom).zip](https://github.com/Sequal32/yourcontrols/files/6185115/JustFlight_PA28_Arrow_III.custom.zip)

**Comment by Sequal32** [comment]
Thanks for this! You did a great job with the definitions as well so I hope you'll be able to contribute to supporting even more planes in the future! Looks like I didn't include a lot of the local variables you mentioned, which I have now done. You can tell me how it goes by using the installer again and trying out it the new definitions :)

**Comment by CptAdhoc** [comment]
I did a small test tonight, on the ground. Door and windows are working now, however we get the door state to desync (opened for one, closed for the other). I didn't tought to test oil and bagage door, but I'm guessing they could benefit of the ResetWhenEquals type.
Flaps and fuel selector are working.
Fuel quantity is working.

We weren't able to continue and fly because throtle, propreler and mixture were not working. They were working with the first beta definition and the C172.yaml, i'm guessing it's due to the removal of engines.yaml include.

**Comment by Sequal32** [comment]
oops I have no clue why I removed that, should be put back now

**Comment by CptAdhoc** [comment]
Taking a wild guess, I think it has something to do with "L:Oil_door" and "L:Baggage_door" :).
Are you looking for something matching for the the "target" ?

We are planning an hour flight tonight with my fellow copilot and I can't find the version where engine.yaml is reintroduced. Maybe it's due to my low Github skills, maybe it's not out yet :D. In the second case, no mean to rush things at all, just wondering if reintroducing engines.yaml is the only thing to do to get the definition ready for the test ;)

**Comment by Sequal32** [comment]
Yeah so the L:Oil_door and the L:Baggage_door are for the animation only. If you try to directly set those variables it'll just reset back to its original position. For the cabin door, there's an LVar that when you set it to 1.0, it'll toggle the door open and trigger the animation but there's no such thing for the oil and the baggage door.

It should be there when you run the beta installer again, but yes it just contains the engines.yaml again so you can add that back manually :)

**Comment by CptAdhoc** [comment]
We did our hour flight, no deal breaking issue :D.
Our major problem were COM1 and NAV1 frequencies on the GNS430 that were not synced, both standby and active. The GNS hitself is working fine, I guess it's affected by all the ignore part of the definition.
Transponder is working, but it seems that some knob movement are skiped sometimes, so you get two different transponder code. We maybe encounter the same problem with the COM2 (KNR175), but further testing has to be done.
The direction indicator seems to desync, I think we need more testing to confirm that point.

No problem on the engine (except the mixture that change sometimes because pilote and copilot use an axis to control that, since everybody can control that, it's logic), no problem on flap, gear, fuel selector and quantity. The heading and ground speed on the GNS430 is fine.



I found nothing more about oil and bagage door. Since the opening is done via the tablet, i'm guessing it's handled by the .wasm file.
